### PR TITLE
[dtests] Added few Negative tests 

### PR DESCRIPTION
**Pull request by satyanveshd**
Added few of the negative scenarios as part of HIP dtests.

### CODE REVIEW

File: tests/src/Negative/memory/hipMemGetInfo.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+/* HIT_START
+ * BUILD: %t %s ../../test_common.cpp EXCLUDE_HIP_PLATFORM nvcc
+ * TEST: %t
+ * HIT_END
+ */
+
+#include "test_common.h"
+#include <iostream>
+#include <hip/hip_runtime.h>
```
Comment (SarbojitAMD) [inline tests/src/Negative/memory/hipMemGetInfo.cpp]: test_common.h has both files included hence can be skipped

File: tests/src/Negative/memory/hipMemGetInfo.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+int main(){
+   hipError_t e;
+   size_t free,total;
+
+   e = hipMemGetInfo(NULL,NULL);
+  
+   HIPASSERT(e!=hipSuccess);
```
Comment (SarbojitAMD) [inline tests/src/Negative/memory/hipMemGetInfo.cpp]: I feel it is good to check exact error than just checking !hipSuccess, e.g. in this case check should be hipErrorInvalidValue

File: tests/src/Negative/memory/hipMemcpyToSymbolAsync.cpp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    HIPCHECK(hipMalloc(&Sd,100));
+
+    hipStream_t stream;
+    HIPCHECK(hipStreamCreate(&stream));
+
+    e = hipMemcpyToSymbolAsync(HIP_SYMBOL(Sd),S,100,0, hipMemcpyHostToDevice, stream);
+
+    HIPCHECK(hipFree(Sd));  
+    HIPASSERT(e!=hipSuccess);
```
Comment (SarbojitAMD) [inline tests/src/Negative/memory/hipMemcpyToSymbolAsync.cpp]: Better to check specific errors

File: tests/src/Negative/memory/hipMemcpyToSymbolAsync.cpp [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    char S[100]="This is not a device symbol";
+    
+    HIPCHECK(hipSetDevice(0));
+
+    HIPCHECK(hipMalloc(&Sd,100));
+
+    hipStream_t stream;
+    HIPCHECK(hipStreamCreate(&stream));
+
+    e = hipMemcpyToSymbolAsync(HIP_SYMBOL(Sd),S,100,0, hipMemcpyHostToDevice, stream);
```
Comment (SarbojitAMD) [inline tests/src/Negative/memory/hipMemcpyToSymbolAsync.cpp]: I think both error cases (hipErrorInvalidValue, hipErrorInvalidSymbol)  should be covered under -ve test cases

**Comment by satyanveshd** [comment]
Raised #1735 after addressing the review comments. So, closing this.

**Pull request closed without merging**